### PR TITLE
DMX controller logic

### PR DESCRIPTION
**Pull request by jakefuentes**
Updating PR to reflect comments and latest from main. Also deleted some old code from pattern.py.

@ben are you aware that the elephant is showing up as all kinds of weird in the visualizer? Looks like that was the case as of #26 -- I saw a bunch of address updates so I'm guessing that was intentional, but it's a little wacky (funky?) right now.

### CODE REVIEW

File: controller/patterns/pattern_config.py [hunk 1]
```diff
@@ -33,4 +31,6 @@
     
     # # Effect patterns
     ('0x2', SparklePattern, dict(color=np.array([255, 255, 255]), sparkle_probability=0.001, decay_param=0.95)),
-]
+    ('1x2', RainbowWavesPattern, dict()),
+    ('1x3', BouncingBlocksPattern, dict())
```
Comment (pitzer) [inline controller/patterns/pattern_config.py:35]: The button assignment should probably by '2x2'. 

Btw the button layout for launchpad is here:
https://github.com/obeezzy/lpminimk3/blob/2041f0ea04594eaf8461e9f884944556b2d9d5ee/lpminimk3/_core/components.py#L784

File: controller/main.py [hunk 2]
```diff
@@ -77,8 +77,9 @@ async def serve(self):
 
 
 class PatternGenerator:
-    def __init__(self, patter_selector):
+    def __init__(self, patter_selector, color_selector=None):
         self.patter_selector = patter_selector
+        self.color_selector = color_selector
```
Comment (pitzer) [inline controller/main.py:82]: Are you planning to add this in the future?

Comment (jakefuentes) [reply controller/main.py:82]: Doh! Old code from an alternate implementation I tried. Removing

File: controller/patterns/pattern_config.py (hunk 1 above)
Comment (jakefuentes) [reply controller/patterns/pattern_config.py:35]: Updated

**Review by pitzer — APPROVED** [review]
Looks great! Please go ahead and merge.

Re elephant looking weird in the visualizer: I checked and it seems fine both in main and this branch. This typically happens when the browser and the light controller use different led configs and it is usually the web page that is stale because it only updates the config when the page is reloaded. So try reloading the page or even restarting the web server to see if that fixes it for you.

**Review by jakefuentes — COMMENTED** [review]
Cleanup

**Pull request merged**